c_src/ep_decoder.c

```diff
@@ -100,6 +100,44 @@ unpack_int32(ErlNifEnv *env, ep_dec_t *dec, ERL_NIF_TERM *term)
     return_error(env, dec->term);
 }
 
+static inline char * toArray(long int number, size_t size)
+{
+    int32_t i;
+    char *numberArray = calloc(size, sizeof(char));
+    for (i = size-1; i >= 0; --i, number /= 10)
+    {
+        numberArray[i] = (number % 10) + '0';
+    }
+    return numberArray;
+}
```

Comment:
C already has a function for this http://www.cplusplus.com/reference/cstdlib/itoa/

Reply:
i know they exist in cpp. but they dont work with my gcc compiler.. so implemented them myself.
it was easier to do it this way and i could see it working for both 32 and 64 bits.
i would prefer to keep it this way for now.